default.nix

```diff
@@ -0,0 +1,50 @@
+#
+# Nix Setup for Hyrise
+#
+# First install Nix (see https://nixos.org/manual/nix/stable/installation/).
+# Then in the root of this repository, run 
+#
+#   nix-shell --pure`
+#
+# The `--pure` flag ensures that the shell does not take over environment
+# variables from the host. Therefore, all software installed comes purely
+# from Nix package manager. 
+# For MacOS user: This also means the default Apple clang is not used ;)
+
+let
+  pkgs = import ./nixpkgs {};
+in
+
+pkgs.stdenv.mkDerivation {
+  name = "hyrise";
+  version = "0.0.1";
+
+  # nativeBuildInput vs. buildInputs
+  #     nativeBuildInputs are packages that are needed to be available during the runtime of the process
+  #     buildInputs are that are needed during the build process, but not anymore during the runtime. 
```

Comment:
build process of Hyrise or nix?

Reply:
I see, the explanation is not clear. I refined the explanation.